packages/rest/src/router/router-spec.ts

```diff
@@ -0,0 +1,34 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/rest
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import {OpenApiSpec} from '@loopback/openapi-v3-types';
+
+export type RouterSpec = Pick<OpenApiSpec, 'paths' | 'components' | 'tags'>;
+
+export function assignRouterSpec(target: RouterSpec, additions: RouterSpec) {
```

Comment:
Similarly here. Please add few unit-level tests to verify how `assignRouteSpec` handles different edge cases. Take a look at `if` conditions and explanatory code comments below to see what to test.